File: week07/explore_phila_data/extract_phl_opa_properties.mjs
import dotenv from 'dotenv';
dotenv.config();

import fetch from 'node-fetch';
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';
import { Storage } from '@google-cloud/storage'

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const DATA_DIR = path.join(__dirname, 'raw_data/');

// Download the OPA Properties data as a CSV
const url = 'https://opendata-downloads.s3.amazonaws.com/opa_properties_public.csv';
const filename = path.join(DATA_DIR, 'phl_opa_properties.csv');

const response = await fetch(url);
if (!response.ok) {
  throw new Error(`HTTP error! status: ${response.status}`);
}

await fs.writeFile(filename, await response.text());

console.log(`Downloaded ${filename}`);

// Upload the downloaded file to cloud storage
const BUCKET_NAME = process.env.DATA_LAKE_BUCKET
const blobname = 'raw/phl_opa_properties/phl_opa_properties.csv';

const storageClient = new Storage();
const bucket = storageClient.bucket(BUCKET_NAME);
await bucket.upload(filename, {
  destination: blobname,
});

console.log(`Uploaded ${filename} to ${BUCKET_NAME}`);

File: week07/explore_phila_data/extract_phl_opa_properties.py
from dotenv import load_dotenv
load_dotenv()

import os
import pathlib
import requests
from google.cloud import storage

DATA_DIR = pathlib.Path(__file__).parent / 'raw_data'

# Download the OPA Properties data as a CSV
url = 'https://opendata-downloads.s3.amazonaws.com/opa_properties_public.csv'
filename = DATA_DIR / 'phl_opa_properties.csv'

response = requests.get(url)
response.raise_for_status()

with open(filename, 'wb') as f:
    f.write(response.content)

print(f'Downloaded {filename}')

# Upload the downloaded file to cloud storage
BUCKET_NAME = os.getenv('DATA_LAKE_BUCKET')
blobname = 'raw/phl_opa_properties/phl_opa_properties.csv'

storage_client = storage.Client()
bucket = storage_client.bucket(BUCKET_NAME)
blob = bucket.blob(blobname)
blob.upload_from_filename(filename)

print(f'Uploaded {blobname} to {BUCKET_NAME}')
